src/main/java/com/linecorp/armeria/server/http/tomcat/TomcatService.java

```diff
@@ -428,9 +435,24 @@ private Request convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage
         final InetSocketAddress localAddr = ctx.localAddress();
         coyoteReq.localAddr().setString(localAddr.getAddress().getHostAddress());
         coyoteReq.localName().setString(hostname);
-        coyoteReq.serverName().setString(hostname);
         coyoteReq.setLocalPort(localAddr.getPort());
-        coyoteReq.setServerPort(localAddr.getPort());
+
+        final String hostHeader = req.headers().get(HttpHeaderNames.AUTHORITY);
+        if (hostHeader == null) {
+            return null;
```

Comment:
But we don't really reject HTTP/1.0 clients and it's probably better being lax against client mistakes? :-)